custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js

```diff
@@ -31,8 +33,12 @@ function MobileFiltersController($scope) {
             $scope.filterData['date'] = data.date;
             $scope.filterData['month'] = data.month;
             $scope.filterData['year'] = data.year;
+        } else if (data.hasAdditionalFilterData) {
+            $scope.filterData['gender'] = data.gender;
+            $scope.filterData['age'] = data.age;
+            $scope.receivedAdditionalFilterData = true;
         }
-        if ($scope.hasLocation && $scope.hasDate) {
+        if ($scope.hasLocation && $scope.hasDate && $scope.receivedAdditionalFilterData ) {
```

Comment:
`receivedAdditionalFilterData`  will always be `true` even if the report doesn't have additional filters? can you explain how that works? it's a bit counter-intuitive to me (again, maybe just a non-intuitive variable name causing the issue)